api/lib/opentelemetry/trace/propagation/job_trace_context_extractor.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+module OpenTelemetry
+  module Trace
+    module Propagation
+      # Extracts context from carriers in the W3C Trace Context format
+      class JobTraceContextExtractor
+        include Context::Propagation::DefaultGetter
+
+        # Returns a new JobTraceContextExtractor that extracts context using the
+        # specified header keys
+        #
+        # @param [String] traceparent_key The traceparent header key used in the carrier
+        # @param [String] tracestate_key The tracestate header key used in the carrier
+        # @return [JobTraceContextExtractor]
+        def initialize(traceparent_key: 'traceparent',
+                       tracestate_key: 'tracestate')
+          @traceparent_key = traceparent_key
+          @tracestate_key = tracestate_key
```

Comment:
I wonder if we really need `tracestate` here? It is typically intended as a representation of vendor-specific trace information as traces traverse systems that may be owned by different companies and instrumented in different ways. That doesn't apply here.

Arguing against myself: the jobs may make HTTP requests to external services that should form part of the same trace, and if we had: A -http-> B -sidekiq-> B -http-> A, then the `tracestate` needs to be propagated back to A.

Thoughts @mwear ?

Reply:
@fbogsany I concur with the argument you pose to yourself. If the job makes a request to another a system that depends on `tracestate` it will break the trace if it isn't sent.